api-gateway/src/main/java/org/swasth/apigateway/filters/HCXValidationFilter.java

```diff
@@ -79,6 +79,9 @@ public GatewayFilter apply(Config config) {
                     jweRequest.validate(getMandatoryHeaders(), getSubject(exchange), timestampRange, senderDetails, recipientDetails);
                     jweRequest.validateUsingAuditData(allowedEntitiesForForward, allowedRolesForForward, senderDetails, recipientDetails, getCorrelationAuditData(jweRequest.getCorrelationId()), getCallAuditData(jweRequest.getApiCallId()), getParticipantCtxAuditData(jweRequest.getSenderCode(), jweRequest.getRecipientCode(), jweRequest.getCorrelationId()), path);
                 } else {
+                    if (!path.contains("on_")) {
+                        throw new ClientException(ErrorCodes.ERR_ACCESS_DENIED, "Does not have access to the called API");
```

Comment:
Use Invalid Payload error and give proper message saying Plain JSON Request is valid only for on_* API calls and it should be proper JWE object for action API calls.